### ISSUE TITLE
DEBAM is broken.

### ISSUE THREAD

**Issue by fmuzf** [regine/meltmodel#47]
Running debam results in segfaults at [writeout.c:2028](https://github.com/regine/meltmodel/blob/1565c2af709c5b122c8c44373d9d0b1d7c183c69/src/writeout.c#L2028).

I think this is a case of the ```meltenergy``` array not being allocated unless the snow model is run.

**Comment by fmuzf** [comment]
Ok, I've merged @chreijmer 's suggested changes in ad24d382284fa0f1b74a3af145840d0a1d143a27.
It works!

**Comment by regine** [comment]
ok,
everything is on master??
How do I download the latest from master if I have a copy on mine and made some changes that I don't want in my files.
the question is how to do it if I don't want to start from scratch, i.e as if i did not have it on my machine?

Thanks
/Regine

On Jan 23, 2015, at 12:06, Lyman Gillispie <notifications@github.com> wrote:

> Ok, I've merged @chreijmer 's suggested changes in ad24d38.
> It works!
> 
> —
> Reply to this email directly or view it on GitHub.
> 



------------------------------------------
Regine Hock
Geophysical Institute
University of Alaska
903 Koyukuk Dr.
Fairbanks, Alaska 99775-7320
ph: 907 - 474 7691
fax: 907 - 474 5882
regine@gi.alaska.edu
------------------------------------------

**Comment by fmuzf** [comment]
I haven't merged this into master yet, I thought it might be better to combine my solutions to this, and issues #45 and #46 into one update/version bump from which everyone can work on.

I think the easiest way to keep from clobbering your changes would be for you to make a new branch off of master, and commit your changes to that branch; when these changes get pulled into master, you can do a merge on your new branch to bring it up to date with master. Something like:

```shell
git checkout -b <topic-branch-name>
< git commands to add/commit your changes>
# time passes and we merge stuff into master
git checkout  master #switch to master
git pull origin #pull down changes from server
git checkout <topic-branch-name>
git merge master #merges the changes made on master into your topic branch
```

Similarly, to test out my changes before I pull  them into master, you can add my repository as a remote on your local version, use git to download my code, and test it out:
```shell
git remote add lyman git@github.com:fmuzf/meltmodel.git #adds my fork as the remote 'lyman'
git fetch lyman #downloads all the stuff i currently have up on gh
git checkout lyman/lyman-issue#46  #my code with the new makefile
```
Now you'll have my code from my branch lyman-issue#46, and you can test things out on it.

To get back to your code/master/whatever, you can just checkout that branch:
```shell
git checkout master
```
and it'll change the files over to those on master.

**Comment by fmuzf** [comment]
This has been fixed in v2.1.2 so I'm closing.
